cardano-api/src/Cardano/Api/Governance/Poll.hs

```diff
@@ -0,0 +1,391 @@
+           _ ->
+             Left (malformedField fieldLbl "Bytes")
+
+
+-- ----------------------------------------------------------------------------
+-- Governance Poll Witness
+--
+
+data GovernancePollWitness
+    = GovernancePollWitnessVRF
+        (VerKeyVRF StandardCrypto)
+        (VRF.CertVRF (VRF StandardCrypto))
+    | GovernancePollWitnessColdKey
+        (VKey 'Witness StandardCrypto)
+        (SignedDSIGN StandardCrypto GovernancePollAnswer)
+  deriving Show
+
+instance HasTypeProxy GovernancePollWitness where
+    data AsType GovernancePollWitness = AsGovernancePollWitness
+    proxyToAsType _ = AsGovernancePollWitness
+
+instance AsTxMetadata GovernancePollWitness where
+    asTxMetadata witness =
+      makeTransactionMetadata $ Map.fromList
+        [ ( metadataLabel
+          , TxMetaMap
+           [ case witness of
+              GovernancePollWitnessVRF vk proof ->
+                ( TxMetaNumber 4
+                , TxMetaList
+                    [ TxMetaBytes $ VRF.rawSerialiseVerKeyVRF vk
+                    , let bytes = VRF.rawSerialiseCertVRF proof in
+                      TxMetaList $
+                        [ TxMetaBytes $ BS.take txMetadataByteStringMaxLength bytes
+                        , TxMetaBytes $ BS.drop txMetadataByteStringMaxLength bytes
+                        ]
+                    ]
+                )
+              GovernancePollWitnessColdKey (VKey vk) (DSIGN.SignedDSIGN sig) ->
+                ( TxMetaNumber 5
+                , TxMetaList
+                    [ TxMetaBytes $ DSIGN.rawSerialiseVerKeyDSIGN vk
+                    , TxMetaBytes $ DSIGN.rawSerialiseSigDSIGN sig
+                    ]
+                )
+           ]
+          )
+        ]
+
+instance SerialiseAsCBOR GovernancePollWitness where
+    serialiseToCBOR =
+      serialiseToCBOR . asTxMetadata
+
+    deserialiseFromCBOR AsGovernancePollWitness bs = do
+      metadata <- deserialiseFromCBOR AsTxMetadata bs
+      withNestedMap lbl metadataLabel metadata $ \values ->
+        case lookup (TxMetaNumber 4) values of
+          Just (TxMetaList [TxMetaBytes vk, TxMetaList[TxMetaBytes proofHead, TxMetaBytes proofTail]]) ->
+            expectJust (field "4") $ GovernancePollWitnessVRF
+              <$> VRF.rawDeserialiseVerKeyVRF vk
+              <*> VRF.rawDeserialiseCertVRF (proofHead <> proofTail)
+          Just _  ->
+            Left $ malformedField (field "4") "List"
+          Nothing ->
+            case lookup (TxMetaNumber 5) values of
+              Just (TxMetaList [TxMetaBytes vk, TxMetaBytes sig]) ->
+                expectJust (field "5") $ GovernancePollWitnessColdKey
+                  <$> fmap VKey (DSIGN.rawDeserialiseVerKeyDSIGN vk)
+                  <*> fmap DSIGN.SignedDSIGN (DSIGN.rawDeserialiseSigDSIGN sig)
+              Just _  ->
+                Left $ malformedField (field "5") "List"
+              Nothing ->
+                Left $ missingField (field "{4|5}")
```

Comment:
This reads to me like we only read `TxMetaNumber 5` if `TxMetaNumber 4` is missing.